common-test/src/main/java/io/strimzi/test/JUnitGroupRule.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.test;
+
+import org.junit.Assume;
+import org.junit.rules.MethodRule;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+public class JUnitGroupRule implements MethodRule {
+
+    @Override
+    public Statement apply(Statement base, FrameworkMethod method, Object target) {
+        JUnitGroup testGroup = method.getAnnotation(JUnitGroup.class);
+
+        if (testGroup == null) {
+            return base;
+        }
+
+        Collection<String> enabledGroups = getEnabledGroups(testGroup.key());
+        Collection<String> declaredGroups = getDeclaredGroups(testGroup);
+
+        if (isGroupEnabled(enabledGroups, declaredGroups)) {
+            return base;
+        }
+
+        return new IgnoreStatement(enabledGroups, declaredGroups);
+    }
+
+    static class IgnoreStatement extends Statement {
```

Comment:
I guess a special statement is necessary because this is implemented as a Rule. 
In contrast the `StrimziRunner` gets to override `isIgnored(FrameworkMethod child)` to determine whether a method is ignored, but maybe it should likewise use `IgnoreStatement`.